From cecae0040aafd0aa5a650febd936947754a1cc08 Mon Sep 17 00:00:00 2001
From: Corey Bryant <corey.bryant@canonical.com>
Date: Fri, 27 Jan 2023 17:06:26 -0500
Subject: [PATCH 37/86] Revert "Bug#34755262 Allow inheritance of
 MySQLRoutingConnectionBase"

This reverts commit 8b4dd31ce98c7ee231aa6a0d832e4bfe7d36b7ac.
---
 router/src/routing/src/classic_connection.h | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/router/src/routing/src/classic_connection.h b/router/src/routing/src/classic_connection.h
index 8886364da96..340726ec722 100644
--- a/router/src/routing/src/classic_connection.h
+++ b/router/src/routing/src/classic_connection.h
@@ -187,9 +187,7 @@ class MysqlRoutingClassicConnection : public MySQLRoutingConnectionBase {
       std::function<void(MySQLRoutingConnectionBase *)> remove_callback)
       : MySQLRoutingConnectionBase{context, std::move(remove_callback)},
         route_destination_{route_destination},
-        destinations_{route_destination_ != nullptr
-                          ? route_destination_->destinations()
-                          : Destinations{}},
+        destinations_{route_destination_->destinations()},
         socket_splicer_{std::make_unique<ProtocolSplicerBase>(
             TlsSwitchableConnection{std::move(client_connection),
                                     std::move(client_routing_connection),
@@ -247,8 +245,7 @@ class MysqlRoutingClassicConnection : public MySQLRoutingConnectionBase {
 
   void server_socket_failed(std::error_code ec, bool call_finish = true);
 
-  virtual void client_socket_failed(std::error_code ec,
-                                    bool call_finish = true);
+  void client_socket_failed(std::error_code ec, bool call_finish = true);
 
  private:
   enum class Function {
-- 
2.37.2

